### PR TITLE
Add Github actions upgrade parent pom

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

File: .github/workflows/maven.yml
Comment (bmarwell) [inline .github/workflows/maven.yml:30]: Do we want to test on different maven versions?

File: src/it/AnnotationProcessingOutput/pom.xml
Comment (bmarwell) [inline src/it/AnnotationProcessingOutput/pom.xml:35]: Lets just upgrade to 8, because otherwise the JDK16 build will fail.

**Review by bmarwell — CHANGES_REQUESTED** [review]
some nits

**Comment by bmarwell** [comment]
WDYT of https://jreleaser.org/? :)

File: .github/workflows/maven.yml
Comment (olamy) [reply .github/workflows/maven.yml:30]: well I would not bother wit that :) 

> WDYT of https://jreleaser.org/? :)

not sure how and why is it better than a simple `mvn release:prepare release:perform -B`?

File: .github/workflows/maven.yml
Comment (bmarwell) [inline .github/workflows/maven.yml:28]: 16? Recommendation is usually: all LTS + latest if the latest is not an LTS.

File: src/main/java/org/codehaus/mojo/aspectj/AbstractAjcCompiler.java
Comment (bmarwell) [inline src/main/java/org/codehaus/mojo/aspectj/AbstractAjcCompiler.java:49]: Why a div?
```suggestion
 * <p>
 * Base class for the two aspectJ compiletime weaving mojos.
 * For all available options see <a href="http://www.eclipse.org/aspectj/doc/released/devguide/ajc-ref.html">ajc-ref</a>
 * </p>
```

File: src/main/java/org/codehaus/mojo/aspectj/AjcCompileMojo.java
Comment (bmarwell) [inline src/main/java/org/codehaus/mojo/aspectj/AjcCompileMojo.java:39]:
```suggestion
 * Weaves all main classes using the AspectJ Compiler Plugin.
```

**Review by bmarwell — COMMENTED** [review]
some more nits

**Comment by kriegaex** [comment]
Sorry, but what is this PR supposed to mean? You said, you are accepting my decision to continue maintaining the plugin separately. But at the same time, you are creating PRs and suddenly after years of inactivity continue committing again in this repository. Are you declaring war? I said, I wanted to avoid a schism and only have one living plugin instance, the others (this one, Nick's one) being retired. Why are you doing this?

**Comment by bmarwell** [comment]
> Are you declaring war?

You really should take a good nap before typing such things.
Forking is not declaring a war. We are not fighting here.

> I said, I wanted to avoid a schism and only have one living plugin instance, the others (this one, Nick's one) being retired. Why are you doing this?

This one was never retired or dead, just inactive. As @olamy already pointed out, you did not write to the mailing list. I am sorry for you that you did not find the mailing lists. Now that there are finally committers paying attention to this plugin, it is not okay for you either.

Please do not discuss political topics on pull requests. I recommend you use the mailing lists for progression ideas.

**Comment by kriegaex** [comment]
So now this is all my fault? This is unfair, after all I have tried and done in the last few years. And what about yourself? First you sent a whole array of PRs to my repository, and now you are turning your back, collaborating here again. Besides, many of the issues addressed in this PR I have already resolved or am in the process of resolving in my own repository.

**Comment by kriegaex** [comment]
I also said in the issue where we were discussing earlier: NO SCHISM! I would rather collaborate here than having two plugin versions out there. But I requested this to be retired and also explained why. I did not even receive a clear reply on my request, but this PR seems to be the reply. You want me not to call this a declaration of war on your part? What else is this "who is not with us, is against us" this supposed to mean? Why can Mojohaus not gracefully acknowledge a fuck-up and retire the plugin in favour of my fork?

**Comment by olamy** [comment]
@kriegaex we proposed to you to collaborate here. 
There are some rules such understanding it's a community project (we try to work as team having fun, not relying on a single person etc...)
So it's your call to accept the rules or not.
I perfectly understand such community aspects might not fit your need.
But the license is MIT so you are free to have a fork and do whatever you want. 
That's the beauty of OpenSource.

**Pull request merged**
